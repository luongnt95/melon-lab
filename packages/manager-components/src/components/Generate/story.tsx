import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react';
import React from 'react';
import Generate from './';

const data = {
  generatedMnemonic:
    'next glass shell collect erupt image drive tumble motor gym glove inside',
  restore: action('restore'),
};

storiesOf('Components|Generate', module).add('Default', () => {
  return <Generate {...data} />;
});
