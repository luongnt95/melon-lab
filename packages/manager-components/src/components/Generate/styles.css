.generate__mnemonic {
  font-size: var(--font-size-xl);
  font-family: monospace;
  font-weight: var(--font-weight-bold);
  margin-top: var(--space-m);
  padding-top: var(--space-s);
  text-align: center;
  border-top: 1px solid color-mod(var(--color-warning) b(10%));
}

.generate__notification {
  margin-bottom: var(--space-m);
}
