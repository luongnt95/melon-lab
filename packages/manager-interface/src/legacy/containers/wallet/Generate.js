import { connect } from 'react-redux';
import { actions as routeActions } from '../../actions/routes';
import Generate from '@melonproject/manager-components/components/GenerateWallet/container';
import { actions as walletAction } from '../../actions/wallet';
import { importWalletFromMnemonic } from '@melonproject/melon.js';

const mapStateToProps = state => ({
  generatedMnemonic: state.wallet.mnemonic,
  restoreWallet: {
    initialValues: {
      mnemonic: '',
    },
  },
  passwordForm: {
    initialValues: {
      password: '',
    },
  },
  error: state.wallet.reason,
});

const getWallet = mnemonic => importWalletFromMnemonic(mnemonic);

const mapDispatchToProps = dispatch => ({
  onSubmitMnemonic: async values => {
    try {
      importWalletFromMnemonic(values.mnemonic);
      return dispatch(walletAction.restoreFromMnemonic(values.mnemonic));
    } catch (err) {
      return dispatch(
        walletAction.restoreFromMnemonicFailed('Invalid BIP39 mnemonic'),
      );
    }
  },
  onSubmitPassword: values => {
    dispatch(walletAction.passwordEntered(values.password));
  },
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Generate);
