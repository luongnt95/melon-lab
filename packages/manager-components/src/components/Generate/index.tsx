import React, { Fragment, StatelessComponent } from 'react';
import Button from '~/blocks/Button';
import Notification from '~/blocks/Notification';

import styles from './styles.css';

export interface GenerateProps {
  generatedMnemonic: string;
  onSubmit;
}

export const Generate: StatelessComponent<GenerateProps> = ({
  generatedMnemonic,
  onSubmit,
}) => {
  return (
    <div className="generate">
      <style jsx>{styles}</style>
      <h3>Generate Account</h3>
      <p>
        The standard bip39 is used to generate a mnemonic phrase, from which
        your wallet will be cryptographically derived.
      </p>
      <div className="generate__notification">
        <Notification isWarning>
          Please write down the following mnemonic and store it in a safe place!
          If you loose your mnemonic you will not be able to access your fund
          again. If someone else gets a copy of this, they can take over your
          wallet & fund and steal your price!
          <div className="generate__mnemonic">{generatedMnemonic}</div>
        </Notification>
      </div>
      <Button style="secondary" onClick={onSubmit}>
        Generate
      </Button>
    </div>
  );
};

export default Generate;
