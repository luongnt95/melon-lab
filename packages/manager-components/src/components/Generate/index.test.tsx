import React from 'react';
import Generate from './index';

const data = {
  restore: () => null,
  mnemonic: 'lorem ipsum',
};

describe('Generate', () => {
  const defaultElement = <Generate {...data} />;
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(defaultElement);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
