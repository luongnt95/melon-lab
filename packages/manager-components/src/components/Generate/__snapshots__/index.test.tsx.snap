// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generate should render correctly 1`] = `
<div
  className="jsx-__scopedHash generate"
>
  <JSXStyle
    css="__scoped"
    styleId="__scopedHash"
  />
  <h3
    className="jsx-__scopedHash"
  >
    Generate Account
  </h3>
  <p
    className="jsx-__scopedHash"
  >
    The standard bip39 is used to generate a mnemonic phrase, from which your wallet will be cryptographically derived.
  </p>
  <div
    className="jsx-__scopedHash generate__notification"
  >
    <Notification
      isWarning={true}
    >
      Please write down the following mnemonic and store it in a safe place! If you loose your mnemonic you will not be able to access your fund again. If someone else gets a copy of this, they can take over your wallet & fund and steal your price!
      <div
        className="jsx-__scopedHash generate__mnemonic"
      />
    </Notification>
  </div>
  <Button
    onClick={[Function]}
    style="secondary"
  >
    Generate
  </Button>
</div>
`;
